lang/cs/Org.Apache.REEF.Client/YARN/IYarnREEFClient.cs

```diff
@@ -33,5 +34,13 @@ public interface IYarnREEFClient : IREEFClient
         /// <returns></returns>
         [Unstable("0.17", "Working in progress for rest API id returned")]
         Task<IReadOnlyDictionary<string, IApplicationReport>> GetApplicationReports();
+
+        /// <summary>
+        /// Kills the job application and return Job status
+        /// </summary>
+        /// <param name="appId"></param>
+        /// <returns></returns>
+        [Unstable("0.17", "Working in progress for rest API status returned")]
+        Task<FinalState> KillJobApplication(string appId);
```

Comment:
I'd prefer for us to make that a bit more user friendly. Maybe catch the exceptions and throw them in a `ApplicationKillFailedException` or so that one can interrogate as to why it failed?